classes/task/poll_stale_conversions.php

```diff
@@ -0,0 +1,123 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * A scheduled task.
+ *
+ * @package    local_smartmedia
+ * @copyright  2019 Matt Porritt <mattp@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace local_smartmedia\task;
+
+use core\task\scheduled_task;
+use local_smartmedia\aws_api;
+use local_smartmedia\aws_elastic_transcoder;
+use \local_smartmedia\conversion;
+
+class poll_stale_conversions extends scheduled_task {
+
+    public function get_name() {
+        return get_string('task:poll_conversions', 'local_smartmedia');
+    }
+
+    public function execute() {
+        mtrace('Starting to poll stale conversions...');
+
+        $records = self::get_stale_conversions();
+        mtrace("Found {count($records)} stale conversions to poll");
+
+        $api = new aws_api();
+        $transcoder = new aws_elastic_transcoder($api->create_elastic_transcoder_client());
+        $conversion = new \local_smartmedia\conversion($transcoder);
+
+        foreach ($records as $record) {
+            self::poll_conversion_status($record, $conversion);
+        }
+
+        mtrace("Finished polling stale conversions");
+    }
+
+    private function get_stale_conversions(): array {
```

Comment:
As it's a fairly gnarly query, it might be worth making this public and having a unit test for this one.

Reply:
Ah, I see you're doing the reflection trick to test. Disregard.

Reply:
It is tested in test_get_stale_conversions